### PR TITLE
Code standards for functional programming

### PR DESCRIPTION
**Pull request by EHER**
Remove just the object-oriented programming rules that are incompatible with the functional programming paradigm

### CODE REVIEW

**Comment by EHER** [comment]
Some other things from #8 could be ignored on code standards.

**Comment by EHER** [comment]
💡 #8 must be merged first

File: tests/functional/pair.phpt [hunk 1]
```diff
@@ -19,7 +19,7 @@ var_dump($first);
 var_dump($second);
 
 --EXPECTF--
-(5, in manus tuas domine commendo spiritum meum)
-(repetitio est mater studiorum, 10)
+(5, "in manus tuas domine commendo spiritum meum")
```
Comment (malukenho) [inline tests/functional/pair.phpt]: This output should not change.
That is the same output I'd have in kotlin.

File: src/Iterable/windowed.php [hunk 2]
```diff
@@ -24,9 +24,9 @@
  *
  * Both {@see $size} and {@see $step} must be positive and can be greater than the number of elements in this collection.
  *
- * @param array $iterable Collection to apply the windows
- * @param int   $size     The number of elements to take in each window
- * @param int   $step     The number of elements to move the window forward by on each step, by default 1
+ * @param mixed[] $iterable Collection to apply the windows
```
Comment (malukenho) [inline src/Iterable/windowed.php]: why not `iterable`?

File: src/Standard/Exception/NotImplementedError.php [hunk 3]
```diff
@@ -9,6 +9,8 @@
 /** @internal */
 final class NotImplementedError extends RuntimeException
 {
-    /** @var string */
-    protected $message = 'An operation is not implemented.';
+    public function __construct(string|null $message = null)
```
Comment (malukenho) [inline src/Standard/Exception/NotImplementedError.php]: I'd prefer to have the previous version. Remove this boilerplate 👍🏼

File: src/Iterable/windowed.php (hunk 2 above)
Comment (EHER) [reply src/Iterable/windowed.php]: Code standard expects to be specific about the type of collections.

File: src/Standard/Exception/NotImplementedError.php (hunk 3 above)
Comment (EHER) [reply src/Standard/Exception/NotImplementedError.php]: phpcs keeps recommending to type the parameter:
```
 13 | ERROR | [x] Property \Codelicia\Niltok\Standard\Exception\NotImplementedError::$message does not have native type hint for its value but it
    |       |     should be possible to add it based on @var annotation "string".
```

but that is not really possible:
```
+Fatal error: Type of Codelicia\Niltok\Standard\Exception\NotImplementedError::$message must not be defined (as in class Exception) in .../github.com/codelicia/niltok/src/Standard/Exception/NotImplementedError.php on line 10
```

File: src/Iterable/windowed.php (hunk 2 above)
Comment (EHER) [reply src/Iterable/windowed.php]: I will remove this check.

File: src/Standard/Exception/NotImplementedError.php (hunk 3 above)
Comment (EHER) [reply src/Standard/Exception/NotImplementedError.php]: I will remove the check.

File: tests/functional/pair.phpt (hunk 1 above)
Comment (EHER) [reply tests/functional/pair.phpt]: I am moving this change to #8

**Review by malukenho — APPROVED** [review]

**Review by malukenho — APPROVED** [review]

**Pull request merged**